### PR TITLE
Added defines.h to simplify the module init function.

### PR DESCRIPTION
**Pull request by AraHaan**
I have added the defines.h into the code, as well as added visual studio files that can be used only to make changes to the code with syntax highlighting.

### CODE REVIEW

**Comment by ZeroIntensity** [comment]
I would rather not have VS solution files on the repository, that seems a bit unnecessary.

File: include/defines.h [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+if (PyType_Ready(&type) < 0) \
+  return NULL
+#define PY_TYPE_ADD_TO_MODULE_OR_RETURN_NULL(name, type) \
+if (PyModule_AddObject(m, Py_STRINGIFY(name), Py_NewRef(&type)) < 0) \
+  return _DecrefModuleAndReturnNULL(m)
+#define PY_ADD_CAPSULE_TO_MODULE_OR_RETURN_NULL(objname, ptr, capsuleName) \
+PyObject *capsule = PyCapsule_New((void *)ptr, capsuleName, NULL); \
+if (capsule == NULL) \
+  return _DecrefModuleAndReturnNULL(m); \
+if (PyModule_AddObject(m, Py_STRINGIFY(name), capsule) < 0) { \
```
Comment (AraHaan) [inline include/defines.h:69]:
```suggestion
if (PyModule_AddObject(m, Py_STRINGIFY(objname), capsule) < 0) { \
```

Comment (AraHaan) [reply include/defines.h]: It seems I forgot to change this part of the macro when I renamed it from ``name`` to ``objname``.

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,7 @@ test.py
 test/
 dist/
 .vscode/
+.vs/
+*.user
+*.sln
+*.vcxproj*
```
Comment (AraHaan) [inline .gitignore:13]: So then the Visual Studio stuffs end up as local only.

**Pull request merged**